src/core/api/notification_policy.go

```diff
@@ -180,6 +181,11 @@ func (w *NotificationPolicyAPI) Put() {
 	policy.ProjectID = w.project.ProjectID
 
 	if err = notification.PolicyMgr.Update(policy); err != nil {
+		if e := orm.AsConflictError(err, "notification policy named %s already exists", policy.Name); e != nil {
```

Comment:
Why don't we call `SendError()` directly here?

Reply:
I think he wants to return special error messages for different errors. But I found that the error message in this handler may be duplicated with the error message in dao. Eg, `failed to update the notification policy: failed to update the notification policy: balabala balabala`

Reply:
already fixed